corehq/apps/hqadmin/reports.py

```diff
@@ -495,3 +502,92 @@ def headers(self):
     @property
     def rows(self):
         return self.table_data.rows
+
+
+class StaleCasesTable:
+    STOP_POINT_DAYS_AGO = 365 * 20
+    AGG_DATE_RANGE = 150
+    STALE_DATE_THRESHOLD_DAYS = 365
+
+    BACKOFF_AMOUNT_DAYS = 30
+    MAX_BACKOFF_COUNT = 2
+
+    def __init__(self):
+        self._rows = None
+        self.stop_date = datetime.now() - timedelta(days=self.STOP_POINT_DAYS_AGO)
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(gettext_lazy("Domain")),
+            DataTablesColumn(gettext_lazy("Case count"))
+        )
+
+    @property
+    def rows(self):
+        if self._rows is None:
+            self._rows = []
+            case_count_by_domain = self._aggregate_case_count_data()
+            for domain, case_count in case_count_by_domain.items():
+                self._rows.append([domain, case_count])
+        return self._rows
+
+    def _aggregate_case_count_data(self):
+        end_date = datetime.now() - timedelta(days=self.STALE_DATE_THRESHOLD_DAYS)
+        agg_res = {}
+        curr_backoff_count = 0
+        curr_agg_date_range = self.AGG_DATE_RANGE
+        domains = self._get_domains()
+        while (True):
+            start_date = end_date - timedelta(days=self.AGG_DATE_RANGE)
+            try:
+                query_res = self._stale_case_count_in_date_range(domains, start_date, end_date)
+            except ESError as e:
+                curr_backoff_count += 1
+                if curr_backoff_count <= self.MAX_BACKOFF_COUNT:
+                    curr_agg_date_range -= self.BACKOFF_AMOUNT_DAYS
+                    curr_backoff_count += 1
```

Comment:
Looks like we are incrementing here it again ?

Reply:
Ah, yeah we should only be incrementing outside the conditional block. Will remove the other one.

Reply:
Addressed in [b228b40](https://github.com/dimagi/commcare-hq/pull/35446/commits/b228b408ee63cb87d5786417b375e517c54afcd6)

Reply:
looks like the wrong one was removed here

Reply:
:man_facepalming: Apologies silly mistake on my end. I did a rebase and put it with the previous commit in [7ec4e5b](https://github.com/dimagi/commcare-hq/pull/35446/commits/7ec4e5b3a68e7c647c8dd70613597a1667d59b8b)

Reply:
No worries